app/controllers/messages_controller.rb

```diff
@@ -2,8 +2,13 @@ class MessagesController < ApplicationController
 	before_action :authenticate_user!
 	before_action :authenticate_control!
```

Comment:
We should update this.  Technically we should authenticate control only for index/show to determine what the user can see, but that can be done inside those methods instead of a blanket authentication.